### PR TITLE
Docker - user users dir for AF files

### LINKED ISSUE #6677: Baseline Scan with Automation Framework not support with OpenShift Security Context
**Describe the bug**
We are using Jenkins to dynamically spawn Jenkins Agent on Kubernetes/OpenShit to run CI/CD and tests.
Openshit has a "Security" Feature that gives a random UserId on the start of the Container/Pod.
This random user has no write access to `/zap/`. The automation framework fails:
```
 2021-07-07 12:29:03,718 Could not find custom hooks file at /var/lib/jenkins/jobs/SDM/SockShop_Deploy/front-end/master/39/.zap_hooks.py 
 2021-07-07 12:29:03,718 Trigger hook: cli_opts, args: 1
 2021-07-07 12:29:03,719 Using port: 37244
 Using the Automation Framework
 Traceback (most recent call last):
   File "/zap/zap-baseline.py", line 629, in <module>
     main(sys.argv[1:])
   File "/zap/zap-baseline.py", line 340, in main
     with open(yaml_file, 'w') as yf:
 PermissionError: [Errno 13] Permission denied: '/zap/zap.yaml'
```
Zap already uses a custom dir `/zap/work` which we are mounting into the container.
Please consider writing everyting directly into this directory.

**To Reproduce**
1. Use OpenShit
2. Start pod with zap
3. Try to run a baseline scan without `--autooff`

**Expected behavior**
zap-baseline.py writes everythin directly to `/zap/wrk`

**Software versions**
 - ZAP: owasp/zap2docker-live:latest

**Additional context**
excerpt from the pod:
```yaml
apiVersion: "v1"
kind: "Pod"
spec:
  containers:
    - command:
        - "cat"
      image: "docker.artifactory.example.com/owasp/zap2docker-live"
      imagePullPolicy: "Always"
      name: "zaproxy"
      resources:
        limits:
          cpu: "250m"
          memory: "1000Mi"
        requests:
          cpu: "50m"
          memory: "100Mi"
      tty: true
      volumeMounts:
        - mountPath: "/zap/wrk"
          name: "zap-volume"
          readOnly: false
        - mountPath: "/var/lib/jenkins"
          name: "workspace-volume"
          readOnly: false
      workingDir: "/var/lib/jenkins"
```

**Would you like to help fix this issue?**
I can help testing a fix on OpenShit3.

### PR DESCRIPTION
**Pull request by psiinon**
Fixes #6677

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: docker/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 All notable changes to the docker containers will be documented in this file.
 
+### 2021-07-08
+ - Changed to use users home directory for the Automation Framework files so it will work for any user
```
Comment (kingthorin) [inline docker/CHANGELOG.md:5]:
```suggestion
 - Changed to use user's home directory for the Automation Framework files so it will work for any user
```

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Oops that should be "comment" not "approve" 🤷

**Comment by psiinon** [comment]
Fixed that :)

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**